### PR TITLE
Adds a batch norm implementation that uses the tfop directly.

### PR DESCRIPTION
**Pull request by bgogul**
This PR defines a @differentiable fusedBatchNorm implementation that uses "tf.FusedBatchNormV2" directly.

### CODE REVIEW

**Comment by bgogul** [comment]
@jekbradbury, can you take a look at this and make sure it is reasonable, please?

**Comment by jph00** [comment]
<p>Thanks @bgogul! Can you please either add a cell to this notebook which runs a training with this, or modify nb 11 to use it, and in either case confirm that you see reasonable speed, memory, and accuracy?</p>

**Comment by lattner** [comment]
Awesome, how much of a speedup does this provide?

**Comment by bgogul** [comment]
> Awesome, how much of a speedup does this provide?
Not a dramatic improvement, unfortunately. I manually changed nb11 to use the TFBatchNorm implemented in this PR.  

batch size: 128
Custom BatchNorm: ~110s,
TF BatchNorm: ~80s

**Comment by bgogul** [comment]
I am just going to merge it in so that we can iterate further.
